diff -durpN gcc-4.6.3.orig/libstdc++-v3/include/bits/atomic_base.h gcc-4.6.3/libstdc++-v3/include/bits/atomic_base.h
--- gcc-4.6.3.orig/libstdc++-v3/include/bits/atomic_base.h	2012-03-12 16:20:01.157339129 -0500
+++ gcc-4.6.3/libstdc++-v3/include/bits/atomic_base.h	2012-03-12 16:22:43.708306967 -0500
@@ -34,7 +34,7 @@
 
 #include <bits/c++config.h>
 #include <stdbool.h>
-#include <stdint.h>
+#include <gstdint.h>
 
 namespace std _GLIBCXX_VISIBILITY(default)
 {
