diff -u -r a/gcc-4.5.2/gcc/config/rs6000/vxworks.h b/gcc-4.5.2/gcc/config/rs6000/vxworks.h
--- a/gcc-4.5.2/gcc/config/rs6000/vxworks.h	2009-10-10 11:43:31.000000000 -0700
+++ b/gcc-4.5.2/gcc/config/rs6000/vxworks.h	2011-05-23 12:48:16.857388980 -0700
@@ -34,6 +34,10 @@
   do						\
     {						\
       builtin_define ("__ppc");			\
+      builtin_define ("__ppc__");		\
+      builtin_define ("__PPC__");		\
+      builtin_define ("__PPC");		\
+      builtin_define ("_WRS_KERNEL");		\
       builtin_define ("__EABI__");		\
       builtin_define ("__ELF__");		\
       builtin_define ("__vxworks");		\
diff -u -r a/gcc-4.5.2/gcc/config/vxlib.c b/gcc-4.5.2/gcc/config/vxlib.c
--- a/gcc-4.5.2/gcc/config/vxlib.c	2009-04-22 03:53:53.000000000 -0700
+++ b/gcc-4.5.2/gcc/config/vxlib.c	2011-05-23 14:39:33.688752289 -0700
@@ -25,9 +25,11 @@
 /* Threads compatibility routines for libgcc2 for VxWorks.
    These are out-of-line routines called from gthr-vxworks.h.  */
 
+#include "config.h"
 #include "tconfig.h"
 #include "tsystem.h"
 #include "gthr.h"
+#include <coretypes.h>
 
 #if defined(__GTHREADS)
 #include <vxWorks.h>
diff -u -r a/gcc-4.5.2/gcc/config/vxlib-tls.c b/gcc-4.5.2/gcc/config/vxlib-tls.c
--- a/gcc-4.5.2/gcc/config/vxlib-tls.c	2009-04-23 03:39:24.000000000 -0700
+++ b/gcc-4.5.2/gcc/config/vxlib-tls.c	2011-05-14 16:54:35.948615728 -0700
@@ -36,6 +36,7 @@
 #include "tconfig.h"
 #include "tsystem.h"
 #include "gthr.h"
+#include "config.h"
 
 #if defined(__GTHREADS)
 #include <vxWorks.h>
diff -u -r a/gcc-4.5.2/gcc/config/vxworks.c b/gcc-4.5.2/gcc/config/vxworks.c
--- a/gcc-4.5.2/gcc/config/vxworks.c	2009-02-20 07:20:38.000000000 -0800
+++ b/gcc-4.5.2/gcc/config/vxworks.c	2011-02-05 12:59:22.000000000 -0800
@@ -69,18 +69,18 @@
   
   *name = get_identifier ("__tls_var");
   
-  field = build_decl (FIELD_DECL, get_identifier ("size"),
+  field = build_decl (DECL_SOURCE_LOCATION (TYPE_NAME (type)), FIELD_DECL, get_identifier ("size"),
 		      unsigned_type_node);
   DECL_CONTEXT (field) = type;
   next_field = field;
 
-  field = build_decl (FIELD_DECL, get_identifier ("module_id"),
+  field = build_decl (DECL_SOURCE_LOCATION (TYPE_NAME (type)), FIELD_DECL, get_identifier ("module_id"),
 		      unsigned_type_node);
   DECL_CONTEXT (field) = type;
   TREE_CHAIN (field) = next_field;
   next_field = field;
 
-  field = build_decl (FIELD_DECL, get_identifier ("offset"),
+  field = build_decl (DECL_SOURCE_LOCATION (TYPE_NAME (type)), FIELD_DECL, get_identifier ("offset"),
 		      unsigned_type_node);
   DECL_CONTEXT (field) = type;
   TREE_CHAIN (field) = next_field;
diff -u -r a/gcc-4.5.2/gcc/gcov-io.c b/gcc-4.5.2/gcc/gcov-io.c
--- a/gcc-4.5.2/gcc/gcov-io.c	2009-11-25 02:55:54.000000000 -0800
+++ b/gcc-4.5.2/gcc/gcov-io.c	2011-02-05 13:42:04.000000000 -0800
@@ -87,7 +87,7 @@
     {
       /* Read-only mode - acquire a read-lock.  */
       s_flock.l_type = F_RDLCK;
-      fd = open (name, O_RDONLY);
+      fd = open (name, O_RDONLY, 0666);
     }
   else
     {
diff -u -r a/gcc-4.5.2/gcc/libgcov.c b/gcc-4.5.2/gcc/libgcov.c
--- a/gcc-4.5.2/gcc/libgcov.c	2009-11-28 08:21:00.000000000 -0800
+++ b/gcc-4.5.2/gcc/libgcov.c	2011-02-05 14:20:14.000000000 -0800
@@ -105,7 +105,7 @@
 
         /* Try to make directory if it doesn't already exist.  */
         if (access (filename, F_OK) == -1
-            && mkdir (filename, 0755) == -1
+            && mkdir (filename) == -1
             /* The directory might have been made by another process.  */
 	    && errno != EEXIST)
 	  {
diff -u -r a/gcc-4.5.2/gcc/system.h b/gcc-4.5.2/gcc/system.h
--- a/gcc-4.5.2/gcc/system.h	2009-12-13 15:00:53.000000000 -0800
+++ b/gcc-4.5.2/gcc/system.h	2011-05-23 14:06:54.992947450 -0700
@@ -399,8 +399,12 @@
 #endif
 
 #ifdef HAVE_MALLOC_H
+#ifdef __VXWORKS__
+#include <stdlib.h>
+#else
 #include <malloc.h>
 #endif
+#endif
 
 #if defined (HAVE_DECL_MALLOC) && !HAVE_DECL_MALLOC
 extern void *malloc (size_t);
diff -u -r a/gcc-4.5.2/libstdc++-v3/config/io/basic_file_stdio.cc b/gcc-4.5.2/libstdc++-v3/config/io/basic_file_stdio.cc
--- a/gcc-4.5.2/libstdc++-v3/config/io/basic_file_stdio.cc	2010-02-16 21:43:24.000000000 -0800
+++ b/gcc-4.5.2/libstdc++-v3/config/io/basic_file_stdio.cc	2011-05-23 14:13:27.328947482 -0700
@@ -38,6 +38,9 @@
 // Pick up ioctl on Solaris 2.8
 #ifdef _GLIBCXX_HAVE_UNISTD_H
 #include <unistd.h>
+#ifdef __VXWORKS__
+#include <ioLib.h>
+#endif
 #endif
 
 // Pick up FIONREAD on Solaris 2
@@ -341,7 +344,13 @@
 #else
     int __num = 0;
 #endif
-    int __r = ioctl(this->fd(), FIONREAD, &__num);
+    int __r = ioctl(this->fd(), FIONREAD,
+#ifndef __VXWORKS__
+    		    &__num
+#else
+    		    (int)&__num
+#endif
+		    );
     if (!__r && __num >= 0)
       return __num; 
 #endif    
diff -u -r a/gcc-4.5.2/libstdc++-v3/include/precompiled/stdc++.h b/gcc-4.5.2/libstdc++-v3/include/precompiled/stdc++.h
--- a/gcc-4.5.2/libstdc++-v3/include/precompiled/stdc++.h	2009-12-21 11:00:34.000000000 -0800
+++ b/gcc-4.5.2/libstdc++-v3/include/precompiled/stdc++.h	2011-05-14 21:13:11.704595030 -0700
@@ -30,6 +30,7 @@
 // 17.4.1.2 Headers
 
 // C
+#include <VxWorks.h>
 #ifndef _GLIBCXX_NO_ASSERT
 #include <cassert>
 #endif
diff -u -r a/gcc-4.5.2/libstdc++-v3/libsupc++/pure.cc b/gcc-4.5.2/libstdc++-v3/libsupc++/pure.cc
--- a/gcc-4.5.2/libstdc++-v3/libsupc++/pure.cc	2009-04-09 08:00:19.000000000 -0700
+++ b/gcc-4.5.2/libstdc++-v3/libsupc++/pure.cc	2011-05-23 14:15:10.737062775 -0700
@@ -29,6 +29,9 @@
 #if _GLIBCXX_HOSTED
 #ifdef _GLIBCXX_HAVE_UNISTD_H
 # include <unistd.h>
+#ifdef __VXWORKS__
+# include <ioLib.h>
+#endif
 # define writestr(str)	write(2, str, sizeof(str) - 1)
 # ifdef __GNU_LIBRARY__
   /* Avoid forcing the library's meaning of `write' on the user program
