diff -durp -x '*~' a/libgcc/libgcov.c b/libgcc/libgcov.c
--- a/libgcc/libgcov.c	2011-01-04 13:05:06.000000000 -0500
+++ b/libgcc/libgcov.c	2012-01-15 17:54:11.566928799 -0500
@@ -119,7 +119,7 @@ create_file_directory (char *filename)
 
         /* Try to make directory if it doesn't already exist.  */
         if (access (filename, F_OK) == -1
-#ifdef TARGET_POSIX_IO
+#if defined(TARGET_POSIX_IO) && !defined(__VXWORKS__)
             && mkdir (filename, 0755) == -1
 #else
             && mkdir (filename) == -1
