diff -u -r a/ioLib.h b/ioLib.h
--- a/ioLib.h	2007-07-24 18:15:34.000000000 -0700
+++ b/ioLib.h	2011-05-14 21:29:31.444995507 -0700
@@ -392,7 +392,11 @@
 extern int	open (const char *, int, int);
 extern int	creat (const char *, int);
 extern int	read (int, char *, size_t);
+#ifdef __cplusplus
+extern int	write (int, const char *, size_t);
+#else
 extern int	write (int, char *, size_t);
+#endif
 extern int 	ioctl (int fd, int function, int arg);
 extern STATUS	close (int);
 extern STATUS	remove (const char *);
diff -u -r a/tool/gnu/ldscripts/link.OUT b/tool/gnu/ldscripts/link.OUT
--- a/tool/gnu/ldscripts/link.OUT	2007-07-24 18:15:42.000000000 -0700
+++ b/tool/gnu/ldscripts/link.OUT	2011-05-14 19:10:10.396484862 -0700
@@ -32,7 +32,7 @@
 01a,29oct01,jab  diab script written
 */
 
-ENTRY(_start)
+/*ENTRY(_start)*/
 SECTIONS
 {
   .text     0 :
