diff -durpN -x '*~' gccdist.orig/WindRiver/vxworks-6.3/target/h/wrn/coreip/net/uio.h gccdist/WindRiver/vxworks-6.3/target/h/wrn/coreip/net/uio.h
--- gccdist.orig/WindRiver/vxworks-6.3/target/h/wrn/coreip/net/uio.h	2012-03-14 23:18:34.975643172 -0500
+++ gccdist/WindRiver/vxworks-6.3/target/h/wrn/coreip/net/uio.h	2012-03-14 23:23:12.590624440 -0500
@@ -82,7 +82,7 @@ struct iovec {
 #define UIO_SMALLIOV	8		/* 8 on stack, else malloc */
 
 #ifdef _WRS_KERNEL
-#include <sys/cdefs.h>
+#include <wrn/coreip/sys/cdefs.h>
 
 enum	uio_rw { UIO_READ, UIO_WRITE };
 
