diff -durpN -x '*~' -x uio.h gccdist.orig/WindRiver/vxworks-6.3/target/h/wrn/coreip/sys/cdefs.h gccdist/WindRiver/vxworks-6.3/target/h/wrn/coreip/sys/cdefs.h
--- gccdist.orig/WindRiver/vxworks-6.3/target/h/wrn/coreip/sys/cdefs.h	2012-03-14 23:18:35.904643107 -0500
+++ gccdist/WindRiver/vxworks-6.3/target/h/wrn/coreip/sys/cdefs.h	2012-03-14 23:24:45.223618185 -0500
@@ -65,7 +65,7 @@ modification history
 #define	__END_DECLS
 #endif
 
-#include <netVersion.h>
+#include <wrn/coreip/netVersion.h>
 
 /*
  * The __CONCAT macro is used to concatenate parts of symbol names, e.g.
