diff -durpN gccdist.orig/WindRiver/vxworks-6.3/target/h/unistd.h gccdist/WindRiver/vxworks-6.3/target/h/unistd.h
--- gccdist.orig/WindRiver/vxworks-6.3/target/h/unistd.h	2012-03-12 20:57:48.772965311 -0500
+++ gccdist/WindRiver/vxworks-6.3/target/h/unistd.h	2012-03-12 20:58:37.137360617 -0500
@@ -46,6 +46,7 @@ extern "C" {
 #endif
 
 #include <vxWorks.h>
+#include <ioLib.h> /* for read/write */
 
 #ifndef SEEK_SET
 #define SEEK_SET           0       /* absolute offset, was L_SET */

