diff -durp -x '*~' -x '*.rej' gccdist.orig/WindRiver/vxworks-6.3/target/h/arch/ppc/regsPpc.h gccdist/WindRiver/vxworks-6.3/target/h/arch/ppc/regsPpc.h
--- gccdist.orig/WindRiver/vxworks-6.3/target/h/arch/ppc/regsPpc.h	2012-03-14 22:08:20.217927604 -0500
+++ gccdist/WindRiver/vxworks-6.3/target/h/arch/ppc/regsPpc.h	2012-03-14 22:09:21.276923474 -0500
@@ -24,6 +24,9 @@ xxxxxxxxxxx,yao  written.
 extern "C" {
 #endif
 
+/* Need obsolete types */
+#include "types.h"
+
 #define GREG_NUM	32	/* has 32 32/64-bit data registers */
 
 #ifndef	_ASMLANGUAGE
