diff --git a/net/uio.h b/net/uio.h
--- /dev/null
+++ b/net/uio.h
@@ -0,0 +1,3 @@
+/* stub file to get gcc to compile, as I don't know how to change the include path for target dirs without mucking up the host dir... */
+#include <wrn/coreip/net/uio.h>
+
diff --git a/netVersion.h b/netVersion.h
--- /dev/null
+++ b/netVersion.h
@@ -0,0 +1,3 @@
+/* stub file to get gcc to compile, as I don't know how to change the include path for target dirs without mucking up the host dir... */
+#include <wrn/coreip/netVersion.h>
+
diff --git a/sys/cdefs.h b/sys/cdefs.h
--- /dev/null
+++ b/sys/cdefs.h
@@ -0,0 +1,3 @@
+/* stub file to get gcc to compile, as I don't know how to change the include path for target dirs without mucking up the host dir... */
+#include <wrn/coreip/sys/cdefs.h>
+
diff --git a/taskLib.h b/taskLib.h
--- a/taskLib.h
+++ b/taskLib.h
@@ -210,7 +210,7 @@ extern "C" {
 #include <vwModNum.h>
 #include <classLib.h>
 #include <qLib.h>
-#include <regs.h>
+#include <vx_regs.h>
 #include <excLib.h>
 #include <private/eventP.h>
 #include <private/semLibP.h>
diff --git a/vx_regs.h b/vx_regs.h
--- /dev/null
+++ b/vx_regs.h
@@ -0,0 +1,3 @@
+/* this is because stupid gcc's configure script gets the include path order ALL wrong */
+#include "./regs.h"
+
