diff -durp gccdist.orig/WindRiver/vxworks-6.3/target/h/arch/ppc/regsPpc.h gccdist/WindRiver/vxworks-6.3/target/h/arch/ppc/regsPpc.h
--- gccdist.orig/WindRiver/vxworks-6.3/target/h/arch/ppc/regsPpc.h	2012-02-12 00:35:27.613086396 -0500
+++ gccdist/WindRiver/vxworks-6.3/target/h/arch/ppc/regsPpc.h	2012-02-12 00:42:50.954543864 -0500
@@ -20,6 +20,9 @@ xxxxxxxxxxx,yao  written.
 #ifndef __INCregsPpch
 #define __INCregsPpch
 
+/* Need Opsolete Types */
+#include "types.h"
+
 #ifdef __cplusplus
 extern "C" {
 #endif
Only in gccdist/WindRiver/vxworks-6.3/target/h/arch/ppc: regsPpc.h~
