--- gccdist.orig/WindRiver/vxworks-6.3/target/h/stdint.h	2007-07-24 18:15:40.000000000 -0400
+++ gccdist/WindRiver/vxworks-6.3/target/h/stdint.h	2012-03-13 11:14:13.659269457 -0400
@@ -1,4 +1,8 @@
 #include <types/vxTypes.h>
 
-typedef int intptr_t;
-typedef unsigned long uintptr_t;
+typedef int32_t intptr_t;
+typedef uint32_t uintptr_t;
+
+typedef int64_t intmax_t;
+typedef uint32_t uintmax_t;
+
+typedef int8_t int_least8_t;
+typedef int16_t int_least16_t;
+typedef int32_t int_least32_t;
+typedef int64_t int_least64_t;
+typedef uint8_t uint_least8_t;
+typedef uint16_t uint_least16_t;
+typedef uint32_t uint_least32_t;
+typedef uint64_t uint_least64_t;
+
+typedef int8_t int_fast8_t;
+typedef int16_t int_fast16_t;
+typedef int32_t int_fast32_t;
+typedef int64_t int_fast64_t;
+typedef uint8_t uint_fast8_t;
+typedef uint16_t uint_fast16_t;
+typedef uint32_t uint_fast32_t;
+typedef uint64_t uint_fast64_t;
+
